django/contrib/admin/views/main.py

```diff
@@ -56,6 +69,11 @@ def __init__(self, request, model, list_display, list_display_links,
         self.sortable_by = sortable_by
 
         # Get search parameters from the query string.
+        _search_form = self.search_form_class(request.GET)
+        if not _search_form.is_valid():
+            for k, error in _search_form.errors.items():
+                messages.error(request, ', '.join(error))
```

Comment:
We don't use keys so maybe shorter:
```python
messages.error(request, ', '.join(_search_form.errors.values()))
```

Reply:
No. You'd have to map via a `str()` call...

```
TypeError: sequence item 0: expected str instance, ErrorList found
```